src/python/pants/backend/scala/goals/repl_ammonite.py

```diff
@@ -0,0 +1,215 @@
+# Copyright 2021 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+import importlib.resources
+import itertools
+
+from pants.core.goals.repl import ReplImplementation, ReplRequest
+from pants.engine.addresses import Addresses
+from pants.engine.fs import (
+    AddPrefix,
+    CreateDigest,
+    Digest,
+    Directory,
+    FileContent,
+    MergeDigests,
+    RemovePrefix,
+    Snapshot,
+)
+from pants.engine.internals.selectors import Get, MultiGet
+from pants.engine.process import BashBinary, Process, ProcessResult
+from pants.engine.rules import collect_rules, rule
+from pants.engine.target import Dependencies, DependenciesRequest
+from pants.engine.unions import UnionRule
+from pants.jvm.classpath import Classpath
+from pants.jvm.compile import ClasspathEntry
+from pants.jvm.jdk_rules import JdkSetup
+from pants.jvm.resolve.coursier_fetch import MaterializedClasspath, MaterializedClasspathRequest
+from pants.jvm.resolve.jvm_tool import JvmToolBase, JvmToolLockfileRequest, JvmToolLockfileSentinel
+from pants.util.docutil import git_url
+from pants.util.logging import LogLevel
+
+
+class AmmoniteSubsystem(JvmToolBase):
+    options_scope = "ammonite"
+
+    # TODO: Get this from a `ScalaToolBase` intermediate base class.
+    SCALA_VERSION = "2.13.6"
```

Comment:
This should probably be derived from the configured scala version. How would doing that interact with having a lock file?